cmd/oras/internal/display/status/interface.go

```diff
@@ -64,6 +64,14 @@ type CopyHandler interface {
 	OnMounted(ctx context.Context, desc ocispec.Descriptor) error
 }
 
+// ManifestPushHandler handles status output for manifest push command.
+type ManifestPushHandler interface {
+	OnManifestExists(desc ocispec.Descriptor) error
+	OnManifestUploading(desc ocispec.Descriptor) error
+	OnManifestUploaded(desc ocispec.Descriptor) error
+	OnManifestPushed(ref string) error
```

Comment:
This should belong to the metadata handler

Reply:
As discussed when doing `oras manifest index create` handlers, this function should be in status handler, as its output is like `Pushed    [registry] test.registry.io/test:v1`. See this comment: https://github.com/oras-project/oras/pull/1509#issuecomment-2425463365

However, we may also need `OnManifestPushed` in metadata handler to output necessary metadata information. Future discussion may be needed.

In `oras attach` and `oras push` commands, this function is currently in metadata handler. We may need to move them to status handler in a future work item. cc: @shizhMSFT

Reply:
We need to discuss more on the effect of tty handlers.

Reply:
Ideal handler workflows:
1. Start operation (status)
2. Process nodes / tasks (status, metadata); Note: For each node, we have status and metadata
   a. Start Processing (status)
   b. Complete processing (status, metadata)
3. Complete operation (status, metadata)

Reply:
Per offline discussion, we don't implement status update for "3. complete operation". Thus, we will drop `OnManifestPushed`. We will still keep `OnManifestUploaded` as the completion of the sub-task/sub-process. For naming convension, it is suggested to rename `OnManifestUploaded` to `OnManifestPushed`.

Reply:
For metadata, we need `OnManifestPushed` (or equivalent) for the record of sub-task/sub-process and `OnComplete` for the record of completion of operation (mainly for rendering if the output format does not support streaming. e.g. JSON).